src/main/scala/com/fulcrumgenomics/util/RefFlatSource.scala

```diff
@@ -0,0 +1,154 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.util
+
+import java.io.{Closeable, File, InputStream}
+
+import com.fulcrumgenomics.util.GeneAnnotations.{Exon, Gene, Transcript}
+import dagr.commons.CommonsDef.PathToFastq
+import htsjdk.samtools.SAMSequenceDictionary
+
+import scala.io.Source
+
+object RefFlatSource {
+  /** Creates a new refflat source from a sequence of lines. */
+  def apply(lines: Seq[String], dict: Option[SAMSequenceDictionary]): RefFlatSource = new RefFlatSource(lines.iterator, dict)
+
+  /** Creates a new fasrefflattq source from an iterator of lines. */
+  def apply(lines: Iterator[String], dict: Option[SAMSequenceDictionary]): RefFlatSource = new RefFlatSource(lines, dict)
+
+  /** Creates a new refflat source from an input stream. */
+  def apply(stream: InputStream, dict: Option[SAMSequenceDictionary]): RefFlatSource = new RefFlatSource(Source.fromInputStream(stream).getLines(), dict, Some(stream))
+
+  /** Creates a new refflat source from an input stream. */
+  def apply(source: Source, dict: Option[SAMSequenceDictionary]): RefFlatSource = new RefFlatSource(source.getLines(), dict, Some(source))
+
+  /** Creates a new refflat source from a File. */
+  def apply(file: File, dict: Option[SAMSequenceDictionary]): RefFlatSource = apply(path=file.toPath, dict=dict)
+
+  /** Creates a new refflat source from a Path. */
+  def apply(path: PathToFastq, dict: Option[SAMSequenceDictionary]): RefFlatSource = apply(Io.toInputStream(path), dict)
+
+  private val Header: String = Seq(
```

Comment:
Yes, Picard uses the latter, which sucks, but it breaks if it doesn't have them that way :/